Mathlib/Algebra/Lie/InvariantForm.lean

```diff
@@ -0,0 +1,218 @@
+      simpa only [LinearMap.BilinForm.isOrtho_def, -- and some default simp lemmas
+        AddSubsemigroup.mem_carrier, AddSubmonoid.mem_toSubsemigroup, Submodule.mem_toAddSubmonoid,
+        LinearMap.BilinForm.mem_orthogonal_iff, LieSubmodule.mem_coeSubmodule]
+    intro H a ha
+    rw [← neg_eq_zero, ← hΦ_inv]
+    exact H _ <| N.lie_mem ha
+
+@[simp]
+lemma orthogonal_toSubmodule (N : LieSubmodule R L M) :
+    (orthogonal Φ hΦ_inv N).toSubmodule = Φ.orthogonal N.toSubmodule := rfl
+
+
+lemma mem_orthogonal (N : LieSubmodule R L M) (y : M) :
+    y ∈ orthogonal Φ hΦ_inv N ↔ ∀ x ∈ N, Φ x y = 0 := by
+  simp [orthogonal, LinearMap.BilinForm.isOrtho_def, LinearMap.BilinForm.mem_orthogonal_iff]
+
+lemma orthogonal_disjoint
+    (Φ : LinearMap.BilinForm R L) (hΦ_nondeg : Φ.Nondegenerate) (hΦ_inv : Φ.lieInvariant L)
+    (hL : ∀ I : LieIdeal R L, IsAtom I → ¬IsLieAbelian I)
+    (I : LieIdeal R L) (hI : IsAtom I) :
+    Disjoint I (orthogonal Φ hΦ_inv I) := by
+  rw [disjoint_iff, ← hI.lt_iff, lt_iff_le_and_ne]
+  suffices ¬I ≤ orthogonal Φ hΦ_inv I by
+    simpa only [inf_le_left, ne_eq, inf_eq_left, true_and]
+  intro contra
+  apply hI.1
+  rw [eq_bot_iff, ← perfect_of_isAtom_of_nonabelian I hI (hL I hI),
+      LieSubmodule.lieIdeal_oper_eq_span, LieSubmodule.lieSpan_le]
+  rintro _ ⟨x, y, rfl⟩
+  simp only [LieSubmodule.bot_coe, Set.mem_singleton_iff]
+  apply hΦ_nondeg
+  intro z
+  rw [hΦ_inv, neg_eq_zero]
+  have hyz : ⁅(x : L), z⁆ ∈ I := lie_mem_left _ _ _ _ _ x.2
+  exact contra hyz y y.2
+
+end ring
+
+section field
+
+variable {K L M : Type*}
+variable [Field K] [LieRing L] [LieAlgebra K L]
+variable [AddCommGroup M] [Module K M] [LieRingModule L M]
+
+variable [Module.Finite K L]
+variable (Φ : LinearMap.BilinForm K L) (hΦ_nondeg : Φ.Nondegenerate)
+variable (hΦ_inv : Φ.lieInvariant L) (hΦ_refl : Φ.IsRefl)
+variable (hL : ∀ I : LieIdeal K L, IsAtom I → ¬IsLieAbelian I)
+
+open FiniteDimensional Submodule in
+lemma orthogonal_isCompl_submodule (I : LieIdeal K L) (hI : IsAtom I) :
+    IsCompl I.toSubmodule (orthogonal Φ hΦ_inv I).toSubmodule := by
+  have := (orthogonal_disjoint Φ hΦ_nondeg hΦ_inv hL I hI).eq_bot
+  apply_fun LieSubmodule.toSubmodule at this
+  simp only [LieSubmodule.inf_coe_toSubmodule, LieSubmodule.bot_coeSubmodule,
+    orthogonal_toSubmodule] at this ⊢
+  refine IsCompl.of_eq this ?_
+  apply (eq_top_of_finrank_eq <| (finrank_le _).antisymm _)
+  conv_rhs => rw [← add_zero (finrank K _)]
+  rw [← finrank_bot K L, ← this, finrank_sup_add_finrank_inf_eq]
+  erw [LinearMap.BilinForm.finrank_add_finrank_orthogonal hΦ_refl (W := I)]
+  exact le_self_add
+
+open FiniteDimensional Submodule in
+lemma orthogonal_isCompl (I : LieIdeal K L) (hI : IsAtom I) :
+    IsCompl I (orthogonal Φ hΦ_inv I) := by
+  have := orthogonal_isCompl_submodule Φ hΦ_nondeg hΦ_inv hΦ_refl hL I hI
+  apply IsCompl.of_eq
+  · rw [eq_bot_iff]
+    exact this.disjoint.le_bot
+  · rw [eq_top_iff]
+    exact this.codisjoint.top_le
```

Comment:
This can be:
```suggestion
  rw [isCompl_iff, LieSubmodule.disjoint_iff_coe_toSubmodule,
    LieSubmodule.codisjoint_iff_coe_toSubmodule, ← isCompl_iff]
  exact orthogonal_isCompl_submodule Φ hΦ_nondeg hΦ_inv hΦ_refl hL I hI
```
but what this really should is that we should add the missing lemma `LieSubmodule.isCompl_iff_coe_toSubmodule` just below `LieSubmodule.codisjoint_iff_coe_toSubmodule`, and then this is a one-liner

Reply:
good point. done